server/src/main/java/io/crate/metadata/GeneratedReference.java

```diff
@@ -244,6 +249,15 @@ public Reference getRelocated(ReferenceIdent referenceIdent) {
         );
     }
 
+    @Override
+    public Reference assignOid(long newOid) {
+        return new GeneratedReference(
+            ref.assignOid(newOid),
+            formattedGeneratedExpression,
+            generatedExpression
+        );
```

Comment:
Initially had a setter for oid and used to directly mutate request refs.

Then decided to be consistent with the current reference implementation: it's immutable, if we want to change smth, we create a clone, like `getRelocated`.

However, there are some drawbacks:
1. Need to create copy per ref - in the `AddColumnRequest` we can have multiple columns (dynamic mapping update adding bunch of columns)
2. Cannot use enhanced loop and had to use  `request.references().set(i, ref.assignOid(currentOID));`
3. Had a bug when I forgot to override assignOid on Index/Geo refs. Added big "copy" ctors just to deal with assign and made SimpeRefs fields protected.  `getRelocated` is also forgotten on Index/Geo ref which basically also leads to losing meta on relocation. I will address this later separately, once I see user facing scenario

should I just drop all this code and let oid be settable?